src/Client.php

```diff
@@ -520,34 +626,32 @@ public function bulk(array $params): ResponseSet
         return $bulk->send();
     }
 
-    /**
-     * Makes calls to the elasticsearch server based on this index.
-     *
-     * It's possible to make any REST query directly over this method
-     *
-     * @param string       $path        Path to call
-     * @param string       $method      Rest method to use (GET, POST, DELETE, PUT)
-     * @param array|string $data        OPTIONAL Arguments as array or pre-encoded string
-     * @param array        $query       OPTIONAL Query params
-     * @param string       $contentType Content-Type sent with this request
-     *
-     * @throws ClientException
-     * @throws ConnectionException
-     * @throws ResponseException
-     */
-    public function request(string $path, string $method = Request::GET, $data = [], array $query = [], string $contentType = Request::DEFAULT_CONTENT_TYPE): Response
```

Comment:
Is there a way to at keep this method at least with the parameters? I expect this to be used a lot.

Reply:
Yes, of course we could keep this method but i think users should start using actions from trait ClientEndpointsTrait. This gives them greater consistency and benefits in further maintaining the application.